### PR TITLE
[DON'T MERGE] Changing links from master to main

### PR DESCRIPTION
**Pull request by danhellem**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-node/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

change default branch to main and need to update all the links to master

# Description of the solution

updated links

### CODE REVIEW

**Comment by danhellem** [comment]
## Do not merge until the afternoon of 11/18

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ This being said, the more you do, the quicker it'll go through our gated build!
 -->
 
 # Checklist
-- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-node/blob/master/.github/CONTRIBUTING.md).
+- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-node/blob/main/.github/CONTRIBUTING.md).
```
Comment (vishnureddy17) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: I know the space was there already but that's not what we want I think.
```suggestion
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-node/blob/main/.github/CONTRIBUTING.md).
```

**Review by vishnureddy17 — APPROVED** [review]

**Comment by anthonyvercolano** [comment]
Do you want to change the links in the horton yaml?

**Review by anthonyvercolano — APPROVED** [review]
Hmmm, we have some samples that refer to a masterKey.  But that is definitely out of scope.

Thanks for this Dan.

**Pull request merged**